provider/netperf_test.py

```diff
@@ -59,11 +59,20 @@ def netperf_stress(test, params, vm):
             logging.info("Netperf test start successfully.")
         else:
             test.error("Can not start netperf client.")
+        start_time = time.time()
         utils_misc.wait_for(lambda: not n_client.is_netperf_running(),
-                            netperf_test_duration, 0, 5,
-                            "Wait netperf test finish %ss" % netperf_test_duration)
+                            test_duration, 0, 5,
+                            "Wait netperf test finish %ss" % test_duration)
+        stop_time = time.time()
+        run_time = stop_time - start_time
+        if test_duration - 5 <= run_time <= test_duration + 10:
+            logging.info("netperf runs successfully")
+        elif test_duration + 10 < run_time:
```

Comment:
This will not happen. The maximum time of it is `netperf_test_duration`, so `stop_time - start_time` should be smaller than netperf_test_duration + an offset time(1~2s).

After utils_misc.wait_for, you also need to check the status of `is_netperf_running`, if the netperf still running, then it maybe hangs.

Reply:
For the first part, can get this interval from the test results, especially for windows guest. 
For the second part, do you mean to change to the following:
stop_time = time.time()
run_time = stop_time - start_time
if n_client.is_netperf_running():
    test.fail("netperf running hangs")
elif test_duration - 5 <= run_time <= test_duration + 10: 
    logging.info("netperf runs successfully") 
elif test_duration + 10 < run_time:
    test.fail("neterf runing timeout,executed %ss" % run_time)
else:
    test.fail("netperf terminated unexpectedly,executed %ss" % run_time)

Reply:
Let me express it more clearly:

1. System requires some time of the netperf execution, so the running time will be a little bit longer than the test time, you can add the offset time into `test_duration`
2. After `wait_for`, we need to check the status of netperf, if it still running, then it hangs.
3. if `run_time` is smaller than `test_duration - 5`, we can think that netperf terminated prematurely.

@luckyh I think I have covered all scenarios of the netperf stress, you can correct me if I missed something else.

Reply:
@PaulYuuu makes sense, and sounds good to me, thanks.

Reply:
@yanglei-rh As discussed above, this shouldn't happen.
`stop_time - start_time` should be a very bit bigger than `test_duration`. The first checkpoint is to check if netperf process still running, the second one is to judge the run_time. Please see the above comment for more details.

Reply:
Hi @PaulYuuu  I have changed it. Please help me review it again. thanks a lot.